### ISSUE TITLE
Clippy "approximate value of `f32::consts::PI` found" in no_std package

### ISSUE THREAD

**Issue by laundmo** [rust-lang/rust-clippy#11705]
### Summary

Clippy suggests using std::f32::consts::PI in no_std crate - this is std, and there is no PI in core::f32

### Lint Name

clippy::approx_constant

### Reproducer

I tried this code:

```rust
pub const PI: f32 = 3.1415926f32;
```

I saw this happen:

```
error: approximate value of `f32::consts::PI` found
  --> src\qmc5883l.rs:28:21
   |
28 | pub const PI: f32 = 3.1415926f32;
   |                     ^^^^^^^^^^^^
   |
   = help: consider using the constant directly
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#approx_constant
   = note: `#[deny(clippy::approx_constant)]` on by default
```

I expected to see this happen:

no warning at all, since this isn't approximate of a available const.

### Version

```text
rustc 1.72.1-nightly (edc37d22d 2023-09-15) (1.72.1.0)
binary: rustc
commit-hash: edc37d22db1cc1b112b91addeb1f79951c58e661
commit-date: 2023-09-15
host: x86_64-pc-windows-msvc
release: 1.72.1-nightly
LLVM version: 16.0.4
```

**Comment by y21** [comment]
I don't see where in this warning it suggests `std` specifically. In no_std, you can use [`core::f32::consts::PI`](https://doc.rust-lang.org/stable/core/f32/consts/constant.PI.html), no?

**Comment by laundmo** [comment]
i apoligize. rust-analyzer wasn't resolving that and i only looked at the `core::f32` docs `constants` section, not the `const` module. my bad.
